content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+### The `subpackages` Section
+As mentioned above, a package may extract parts of its contents into subpackages, in order to make for a slimmer final apk. Many packages have resources that are not required at execution time: development headers, man pages, shared libraries that are optional… you name it. This part is really important in Wolfi, because we want packages to be minimal. The `subpackages` section of the melange YAML file looks a lot like the pipeline section, and it essentially works the same way. You'll just have to make sure you place any subpackage files in the `targets.subpkgdir` location.
+
+The `split` built-in pipelines were created to facilitate the creation of subpackages. They implement code to remove development headers (`split/dev`), man pages (`split/manpages`), among other resources that aren't typically required at runtime. You can experiment with those, just be aware that they use standard path locations and some compiled packages may use different paths for certain resources.
+
+For example, this is how a step in the subpackages section would be written, using the `split/dev` built-in pipeline to generate the `php-dev` subpackage:
+
+```yaml
+  - name: php-dev
+    description: PHP 8.2 development headers
+    pipeline:
+      - uses: split/dev
+```
+
+#### Looping with Ranges
+In some cases, you may find yourself repeating the same task over and over with just a couple different values (such as package names). In such scenarios, you can define a range of data that you can "loop" through in a step. For example, let's have a look at how the PHP package uses this feature to create its subpackages.
+
+First, we define an `extensions` range. This should go on a `data` node at the same level as the `pipelines` section of your YAML:
+
+```yaml
+data:
+  - name: extensions
+    items:
+      bz2: Bzip2
+      curl: cURL
+      gd: GD imaging
+      gmp: GNU GMP support
+      ldap: LDAP
+      mysqlnd: MySQLnd
+      openssl: OpenSSL
+      pdo_mysql: MySQL driver for PDO
+      pdo_sqlite: SQLite 3.x driver for PDO
+      soap: SOAP
+      sodium: Sodium
+      calendar: Calendar
+```
+
+In the subpackages section, we define a pipeline for that range:
+
+```yaml
+  - range: extensions
+    name: "php-${{range.key}}"
+    description: "The ${{range.value}} extension"
+    pipeline:
+      - runs: |
+        export EXTENSIONS_DIR=usr/lib/php/modules
+        export CONF_DIR="${{targets.subpkgdir}}/etc/php/conf.d"
+        mkdir -p "${{targets.subpkgdir}}"/$EXTENSIONS_DIR $CONF_DIR
+        mv "${{targets.destdir}}/$EXTENSIONS_DIR/${{range.key}}.so" \
+          "${{targets.subpkgdir}}/$EXTENSIONS_DIR/${{range.key}}.so"
+        prefix=
+        [ "${{range.key}}" != "opcache" ] || prefix="zend_"
+        echo "${prefix}extension=${{range.key}}.so" > $CONF_DIR/"${{range.key}}.ini"
+```
+
+And this will loop through all values of the `extensions` range and execute the described pipeline.
+
+#### The `update` Section
```

Comment:
```suggestion
### The `update` Section
```

Not sure, but it seems like it's supposed to be an h3?

Reply:
good catch!